shared/src/native-src/loader.cpp

```diff
@@ -324,24 +321,136 @@ namespace shared
+                             ", AssemblyID=" + assemblyIdHex +
+                             ", AssemblyName=" + ToString(assemblyNameString) +
+                             ", AppDomainID=" + appDomainIdHex +
+                             ", methodDef=" + HexStr(methodDef) +
+                             ", loaderMethodDef=" + HexStr(loaderMethodDef) +
+                             "]");
+                    }
+                }
             }
 
-            auto enumMethods = EnumMethodsWithName(metadataImport, appDomainTypeDef, SpecificMethodToInjectName);
-            auto enumIterator = enumMethods.begin();
-            if (enumIterator != enumMethods.end()) {
-                auto methodDef = *enumIterator;
+            return hr;
+        }
+
+        if (_loaderOptions.RewriteModulesEntrypoint)
+        {
+            //
+            // Rewrite Module EntryPoint
+            //
+            const mdToken moduleEntryPoint = GetModuleEntryPointToken(moduleBaseLoadAddress, moduleFlags);
+            if (moduleEntryPoint != NULL && moduleEntryPoint != mdTokenNil && _corlibMetadata.Token != mdAssemblyNil)
+            {
+                constexpr DWORD NameBuffSize = 1024;
+
+                mdToken moduleEntryPointParent;
+                WCHAR moduleEntryPointName[NameBuffSize]{};
+                DWORD moduleEntryPointNameLength = 0;
+                hr = metadataImport->GetMemberProps(moduleEntryPoint, &moduleEntryPointParent, moduleEntryPointName,
+                                                    NameBuffSize, &moduleEntryPointNameLength, nullptr, nullptr,
+                                                    nullptr, nullptr, nullptr,
+                                                    nullptr, nullptr, nullptr);
+                if (FAILED(hr))
+                {
+                    Error("Loader::InjectLoaderToModuleInitializer: Call to GetMemberProps(..) returned a FAILED "
+                          "HResult: " + ToString(hr) + ".");
+                    return S_FALSE;
+                }
+
+                WCHAR typeName[NameBuffSize]{};
+                DWORD typeNameLength = 0;
+                DWORD typeFlags;
+                hr = metadataImport->GetTypeDefProps(moduleEntryPointParent, typeName, NameBuffSize, &typeNameLength,
+                                                     &typeFlags, NULL);
+                if (FAILED(hr))
+                {
+                    Error("Loader::InjectLoaderToModuleInitializer: Call to GetTypeDefProps(..) returned a FAILED "
+                          "HResult: " + ToString(hr) + ".");
+                    return S_FALSE;
+                }
+
+                const std::string moduleEntryPointHex = HexStr(moduleEntryPoint);
+                const std::string moduleEntryPointFullName = ToString(typeName) + "." + ToString(moduleEntryPointName);
+
+                Debug("Loader::InjectLoaderToModuleInitializer: Module entrypoint found at: " + moduleEntryPointHex +
```

Comment:
Don't we need to check if (_loaderOptions.LogDebugIsEnabled) here?

Reply:
Yes you're right, done in 742ab68